collect_app/src/main/java/org/odk/collect/android/activities/viewmodels/FormMapViewModel.java

```diff
@@ -99,6 +102,15 @@ private void initializeFormInstances() {
         return mappableFormInstances;
     }
 
+    public MappableFormInstance getInstanceById(long instanceId) {
+        for (MappableFormInstance mappableFormInstance : mappableFormInstances) {
```

Comment:
Doing a sequential search here could be slow if there are lots of instances. I might have considered saving the database id to make this lookup fast. It's probably not that important to do now but wanted to flag it for future consideration.

Reply:
It is removed after removing BottomSheetDialogFragment